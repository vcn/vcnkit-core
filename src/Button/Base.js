import React, { useCallback } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const Base = ({ className, onClick, href, disabled, children, type = 'button', ...rest }) => {

    const handleClick = useCallback(
        (event, ...rest) => {
            if (!onClick) {
                return;
            }

            if (disabled) {
                event.preventDefault();
                return;
            }

            onClick(event, ...rest);
        },
        [ disabled, onClick ]
    );

    if (href) {
        return (
            <a
                className={ className }
                onClick={ handleClick }
                href={ href }
                aria-disabled={ disabled || undefined }
                { ...rest }
            >
                { children }
            </a>
        );
    }

    return (
        <button
            className={ className }
            onClick={ handleClick }
            disabled={ disabled }
            type={ type }
            { ...rest }
        >
            { children }
        </button>
    );
};

Base.propTypes = {
    className: PropTypes.string,
    onClick:   PropTypes.func,
    href:      PropTypes.string,
    disabled:  PropTypes.bool,
    type:      PropTypes.oneOf([ 'button', 'submit' ]),
    children:  PropTypes.node.isRequired,
};

export default styled(Base)`
    background: transparent;
    
    display:         flex;
    align-items:     center;
    justify-content: center;

    font-size:       .8125rem;
    font-weight:     500;
    text-decoration: none;
    text-transform:  uppercase;
    white-space:     nowrap;

    padding:       0 .5rem;
    min-width:     4rem;
    min-height:    2rem;
    border:        none;
    border-radius: .125rem;

    ${ props => `
        font-family: ${ (props.theme.vcnkit && props.theme.vcnkit.fontFamily) || 'Roboto, sans-serif' };
        color:       ${ (props.theme.vcnkit && props.theme.vcnkit.textColor) || 'hsla(0, 0%, 0%, .87)' };
    ` }
    
    &::-moz-focus-inner {
        border: none;
    }

    &:focus {
        outline: none;
    }
    
    &:disabled, &[aria-disabled] {
        color:  ${ props => (props.theme.vcnkit && props.theme.vcnkit.disabledColor) || 'hsla(0, 0%, 0%, .12)' };
        cursor: not-allowed;
    }

    &:not(:disabled):not([aria-disabled]) {
        cursor: pointer;

        &:hover, &:focus {
            background: ${ props => (props.theme.vcnkit && props.theme.vcnkit.focusColor) || 'hsla(0, 0%, 0%, .06)' };
        }
    }
`;
