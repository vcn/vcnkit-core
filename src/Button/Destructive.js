import styled from 'styled-components';

import Base from './Base';

export default styled(Base)`
    color: ${ props => (props.theme.vcnkit && props.theme.vcnkit.destructiveColor) || '#E53935' };
`;
