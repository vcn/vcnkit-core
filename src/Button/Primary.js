import styled from 'styled-components';

import Base from './Base';

export default styled(Base)`
    color: ${ props => (props.theme.vcnkit && props.theme.vcnkit.primaryColor) || '#5282B5' };
`;
