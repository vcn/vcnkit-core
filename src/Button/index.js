export { default as Primary } from './Primary';
export { default as Destructive } from './Destructive';