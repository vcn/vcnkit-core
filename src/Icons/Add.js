import * as React from 'react';

/**
 * @copyright Google
 * @see https://github.com/google/material-design-icons
 * @license Apache-2.0
 */
export default props => (
    <svg
        width="24"
        height="24"
        viewBox="0 0 24 24"
        { ...props }
    >
        <path
            d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
        />
        <path
            d="M0 0h24v24H0z"
            fill="none"
        />
    </svg>
);
