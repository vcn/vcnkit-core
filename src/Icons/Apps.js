import * as React from 'react';

/**
 * @copyright Google
 * @see https://github.com/google/material-design-icons
 * @license Apache-2.0
 */
export default props => (
    <svg
        width="24"
        height="24"
        viewBox="0 0 24 24"
        { ...props }
    >
        <path
            d="M4 8h4V4H4v4zm6 12h4v-4h-4v4zm-6 0h4v-4H4v4zm0-6h4v-4H4v4zm6 0h4v-4h-4v4zm6-10v4h4V4h-4zm-6 4h4V4h-4v4zm6 6h4v-4h-4v4zm0 6h4v-4h-4v4z"
        />
        <path
            d="M0 0h24v24H0z"
            fill="none"
        />
    </svg>
);
