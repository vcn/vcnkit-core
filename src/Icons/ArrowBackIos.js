import * as React from 'react';

/**
 * @copyright Google
 * @see https://github.com/google/material-design-icons
 * @license Apache-2.0
 */
export default props => (
    <svg
        width="24"
        height="24"
        viewBox="0 0 24 24"
        { ...props }
    >
        <path
            d="M11.67 3.87L9.9 2.1 0 12l9.9 9.9 1.77-1.77L3.54 12z"
        />
        <path
            fill="none"
            d="M0 0h24v24H0z"
        />
    </svg>
);
