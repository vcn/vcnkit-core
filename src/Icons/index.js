export { default as Add } from './Add';
export { default as Apps } from './Apps';
export { default as ArrowBack } from './ArrowBack';
export { default as ArrowBackIos } from './ArrowBackIos';
export { default as ArrowForwardIos } from './ArrowForwardIos';
export { default as AttachFile } from './AttachFile';
export { default as Business } from './Business';
export { default as Close } from './Close';
export { default as Clear } from './Clear';
export { default as Check } from './Check';
export { default as ContactMail } from './ContactMail';
export { default as ContactPhone } from './ContactPhone';
export { default as Download } from './Download';
export { default as Email } from './Email';
export { default as Flask } from './Flask';
export { default as Feedback } from './Feedback';
export { default as DesktopWindows } from './DesktopWindows';
export { default as Info } from './Info';
export { default as Inbox } from './Inbox';
export { default as KeyboardArrowDown } from './KeyboardArrowDown';
export { default as KeyboardArrowUp } from './KeyboardArrowUp';
export { default as KeyboardArrowRight } from './KeyboardArrowRight';
export { default as Money } from './Money';
export { default as MarkUnreadMailbox } from './MarkUnreadMailbox';
export { default as Message } from './Message';
export { default as OpenInNew } from './OpenInNew';
export { default as Pdf } from './Pdf';
export { default as Person } from './Person';
export { default as Phone } from './Phone';
export { default as Place } from './Place';
export { default as Poll } from './Poll';
export { default as Send } from './Send';
export { default as Search } from './Search';
export { default as SettingsApplications } from './SettingsApplications';
export { default as Style } from './Style';
export { default as Security } from './Security';
export { default as SupervisedUserCircle } from './SupervisedUserCircle';
export { default as VpnKey } from './VpnKey';
export { default as Warning } from './Warning';