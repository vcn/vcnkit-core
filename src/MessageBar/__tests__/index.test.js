import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import 'jest-styled-components';

import * as Button from '../../Button';
import { Feedback } from '../../Icons';
import * as MessageBar from '../';

describe('MessageBar', () => {
    test('Normal message renders correctly', () => {
        const tree = renderer.create(
            <MessageBar.Message>
                MessageMessageBar
            </MessageBar.Message>
        ).toJSON();
    
        expect(tree).toMatchSnapshot();
    });
    
    test('Info message renders correctly', () => {
        const tree = renderer.create(
            <MessageBar.Info>
                InfoMessageBar
            </MessageBar.Info>
        ).toJSON();
    
        expect(tree).toMatchSnapshot();
    });

    test('Warning message renders correctly', () => {
        const tree = renderer.create(
            <MessageBar.Warning>
                WarningMessageBar
            </MessageBar.Warning>
        ).toJSON();
    
        expect(tree).toMatchSnapshot();
    });

    test('Error message renders correctly', () => {
        const tree = renderer.create(
            <MessageBar.Error>
                ErrorMessageBar
            </MessageBar.Error>
        ).toJSON();
    
        expect(tree).toMatchSnapshot();
    });

    test('Experimental message renders correctly', () => {
        const tree = renderer.create(
            <MessageBar.Experimental>
                ExperimentalMessageBar
            </MessageBar.Experimental>
        ).toJSON();
    
        expect(tree).toMatchSnapshot();
    });

    test('Renders correctly with a different icon', () => {   
        const messageBar = mount(
            <MessageBar.Message
                icon={ <Feedback /> }
            >
                MessageMessageBar
            </MessageBar.Message>           
        );

        expect(messageBar.find(Feedback).length).toBe(1);
    });

    test('Renders correctly with an action', () => {
        const messageBar = mount(
            <MessageBar.Message
                actions={ <Button.Primary>Test</Button.Primary> }
            >
                MessageMessageBar
            </MessageBar.Message>           
        );

        expect(messageBar.find(Button.Primary).length).toBe(1);
    });
});
