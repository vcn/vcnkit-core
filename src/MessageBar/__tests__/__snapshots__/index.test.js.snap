// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageBar Error message renders correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: .8125rem;
  padding: .625rem 1.5rem;
  border-radius: .125rem;
  min-height: 2rem;
  background: hsla(0,0%,0%,.03);
  font-family: "Roboto",sans-serif;
  color: hsla(0,0%,0%,.67);
  background: #C26565;
  color: hsla(0,100%,100%,.87);
}

.c0 svg {
  width: 1.5rem;
  height: 1.5rem;
  margin: 0 1rem 0 0;
  -webkit-flex: 0 0 1.5rem;
  -ms-flex: 0 0 1.5rem;
  flex: 0 0 1.5rem;
}

.c0 svg {
  fill: hsla(0,0%,0%,.57);
}

.c0 svg {
  fill: #FFFFFF;
}

.c0 .c1 {
  color: #FFFFFF;
}

<div
  className="c0"
>
  <svg
    height="24"
    viewBox="0 0 24 24"
    width="24"
  >
    <path
      d="M0 0h24v24H0z"
      fill="none"
    />
    <path
      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
    />
  </svg>
  <div>
    ErrorMessageBar
  </div>
</div>
`;

exports[`MessageBar Experimental message renders correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: .8125rem;
  padding: .625rem 1.5rem;
  border-radius: .125rem;
  min-height: 2rem;
  background: hsla(0,0%,0%,.03);
  font-family: "Roboto",sans-serif;
  color: hsla(0,0%,0%,.67);
  background: #7E57C2;
  color: hsla(0,100%,100%,.87);
}

.c0 svg {
  width: 1.5rem;
  height: 1.5rem;
  margin: 0 1rem 0 0;
  -webkit-flex: 0 0 1.5rem;
  -ms-flex: 0 0 1.5rem;
  flex: 0 0 1.5rem;
}

.c0 svg {
  fill: hsla(0,0%,0%,.57);
}

.c0 svg {
  fill: #FFFFFF;
}

.c0 .c1 {
  color: #FFFFFF;
}

<div
  className="c0"
>
  <svg
    height="24"
    viewBox="0 0 24 24"
    width="24"
  >
    <path
      d="M5,19A1,1 0 0,0 6,20H18A1,1 0 0,0 19,19C19,18.79 18.93,18.59 18.82,18.43L13,8.35V4H11V8.35L5.18,18.43C5.07,18.59 5,18.79 5,19M6,22A3,3 0 0,1 3,19C3,18.4 3.18,17.84 3.5,17.37L9,7.81V6A1,1 0 0,1 8,5V4A2,2 0 0,1 10,2H14A2,2 0 0,1 16,4V5A1,1 0 0,1 15,6V7.81L20.5,17.37C20.82,17.84 21,18.4 21,19A3,3 0 0,1 18,22H6M13,16L14.34,14.66L16.27,18H7.73L10.39,13.39L13,16M12.5,12A0.5,0.5 0 0,1 13,12.5A0.5,0.5 0 0,1 12.5,13A0.5,0.5 0 0,1 12,12.5A0.5,0.5 0 0,1 12.5,12Z"
    />
  </svg>
  <div>
    ExperimentalMessageBar
  </div>
</div>
`;

exports[`MessageBar Info message renders correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: .8125rem;
  padding: .625rem 1.5rem;
  border-radius: .125rem;
  min-height: 2rem;
  background: hsla(0,0%,0%,.03);
  font-family: "Roboto",sans-serif;
  color: hsla(0,0%,0%,.67);
  background: #D3DFEC;
}

.c0 svg {
  width: 1.5rem;
  height: 1.5rem;
  margin: 0 1rem 0 0;
  -webkit-flex: 0 0 1.5rem;
  -ms-flex: 0 0 1.5rem;
  flex: 0 0 1.5rem;
}

.c0 svg {
  fill: hsla(0,0%,0%,.57);
}

<div
  className="c0"
>
  <svg
    height="24"
    viewBox="0 0 24 24"
    width="24"
  >
    <path
      d="M0 0h24v24H0z"
      fill="none"
    />
    <path
      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
    />
  </svg>
  <div>
    InfoMessageBar
  </div>
</div>
`;

exports[`MessageBar Normal message renders correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: .8125rem;
  padding: .625rem 1.5rem;
  border-radius: .125rem;
  min-height: 2rem;
  background: hsla(0,0%,0%,.03);
  font-family: "Roboto",sans-serif;
  color: hsla(0,0%,0%,.67);
}

.c0 svg {
  width: 1.5rem;
  height: 1.5rem;
  margin: 0 1rem 0 0;
  -webkit-flex: 0 0 1.5rem;
  -ms-flex: 0 0 1.5rem;
  flex: 0 0 1.5rem;
}

.c0 svg {
  fill: hsla(0,0%,0%,.57);
}

<div
  className="c0"
>
  <svg
    height="24"
    viewBox="0 0 24 24"
    width="24"
  >
    <path
      d="M0 0h24v24H0z"
      fill="none"
    />
    <path
      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
    />
  </svg>
  <div>
    MessageMessageBar
  </div>
</div>
`;

exports[`MessageBar Warning message renders correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: .8125rem;
  padding: .625rem 1.5rem;
  border-radius: .125rem;
  min-height: 2rem;
  background: hsla(0,0%,0%,.03);
  font-family: "Roboto",sans-serif;
  color: hsla(0,0%,0%,.67);
  background: #EFE1B3;
}

.c0 svg {
  width: 1.5rem;
  height: 1.5rem;
  margin: 0 1rem 0 0;
  -webkit-flex: 0 0 1.5rem;
  -ms-flex: 0 0 1.5rem;
  flex: 0 0 1.5rem;
}

.c0 svg {
  fill: hsla(0,0%,0%,.57);
}

<div
  className="c0"
>
  <svg
    height="24"
    viewBox="0 0 24 24"
    width="24"
  >
    <path
      d="M0 0h24v24H0z"
      fill="none"
    />
    <path
      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
    />
  </svg>
  <div>
    WarningMessageBar
  </div>
</div>
`;
